src/plugins/console/public/application/containers/editor/monaco/monaco_editor_actions_provider.ts

```diff
@@ -340,4 +342,47 @@ export class MonacoEditorActionsProvider {
   ): monaco.languages.ProviderResult<monaco.languages.CompletionList> {
     return this.getSuggestions(model, position);
   }
+
+  private getTextInRange(selectionRange: monaco.IRange): string {
+    const model = this.editor.getModel();
+    if (!model || !selectionRange) {
+      return '';
+    }
+    const { startLineNumber, startColumn, endLineNumber, endColumn } = selectionRange;
+    const text = model.getValueInRange({
+      startLineNumber,
+      startColumn,
+      endLineNumber,
+      endColumn,
+    });
+    return text;
+  }
+
+  public async autoIndent(event: React.MouseEvent) {
+    event.preventDefault();
+    const parsedRequests = await this.getSelectedParsedRequests();
+    const selectionStartLineNumber = parsedRequests[0].startLineNumber;
+    const selectionEndLineNumber = parsedRequests[parsedRequests.length - 1].endLineNumber;
+    const selectedRange = new monaco.Range(
+      selectionStartLineNumber,
+      1,
+      selectionEndLineNumber,
+      this.editor.getModel()?.getLineMaxColumn(selectionEndLineNumber) ?? 1
+    );
+
+    if (parsedRequests.length < 1) {
+      return;
+    }
+
+    const selectedText = this.getTextInRange(selectedRange);
+
+    const autoIndentedText = getAutoIndentedRequests(parsedRequests, selectedText);
+
+    this.editor.executeEdits('', [
```

Comment:
Ah I wasn't sure how this string would be used... will it be visible to the user somehow or do you think we'll only need it for internal use? I added a translated string in [50363cc](https://github.com/elastic/kibana/pull/181613/commits/50363cc7a497d97823e9fe92cded2727a299fcc6) (in case it will be displayed to the user), let me know what you think.

Reply:
I think the string might be used for undo/redo actions which I don't think we currently have. I believe the value won't be displayed in the UI to the user though.

Reply:
I see... Then it might be good to store this in a constant in case it's needed somewhere. I added this change with [3e1264f](https://github.com/elastic/kibana/pull/181613/commits/3e1264f4f8718c48d1c590192c5115afca3f0bb7).